lms/djangoapps/course_api/blocks/forms.py

```diff
@@ -106,20 +108,51 @@ def clean(self):
         cleaned_data['user'] = self._clean_requested_user(cleaned_data, usage_key.course_key)
         return cleaned_data
 
+    def clean_username(self):
+        if 'username' in self.data:
+            return self.cleaned_data['username']
+        else:
+            return None
+
     def _clean_requested_user(self, cleaned_data, course_key):
         """
         Validates and returns the requested_user, while checking permissions.
         """
         requesting_user = self.initial['requesting_user']
         requested_username = cleaned_data.get('username', None)
 
-        if not requested_username:
+        if requesting_user.is_anonymous:
+            return self._verify_anonymous_user(cleaned_data, course_key)
+
+        if requested_username is None:
             return self._verify_no_user(requesting_user, cleaned_data, course_key)
         elif requesting_user.username.lower() == requested_username.lower():
             return self._verify_requesting_user(requesting_user, course_key)
         else:
             return self._verify_other_user(requesting_user, requested_username, course_key)
 
+    @staticmethod
+    def _verify_anonymous_user(cleaned_data, course_key):
+        """
+        Verifies form for when the requesting user is anonymous.
+        """
+        if cleaned_data.get('all_block', None) is not None:
+            raise PermissionDenied(
+                u"Anonymous users do not have permission to access all blocks in '{course_key}'."
+                    .format(course_key=six.text_type(course_key))
```

Comment:
@xitij2000, can you fix the code quality errors reported on this line and 2 other occurrences of the same issue in the indentation of the line containing `.format`?